ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/metrics_correlation/MetricsCorrelation.java

```diff
@@ -0,0 +1,306 @@
+                        // load the model in memory
+                        deployModel(modelInfo.get(MLModel.MODEL_ID_FIELD).toString(), ActionListener.wrap(deployModelResponse -> modelId = getTask(deployModelResponse.getTaskId()).getModelId(), e -> log.error("Metrics correlation model didn't get deployed to the index successfully", e)));
+                    }
+                }
+            }, e -> {
+                //If the model index didn't get created before this request then we can face model index not found exception
+                log.error("Model Index Not found", e);
+                try {
+                    registerModel(ActionListener.wrap(registerModelResponse ->
+                                    modelId = getTask(registerModelResponse.getTaskId()).getModelId(),
+                            ex -> log.error("Exception during registering the Metrics correlation model", ex)));
+                } catch (InterruptedException ex) {
+                    throw new RuntimeException(ex);
+                }
+            }));
+        } else {
+            MLModel model = getModel(modelId);
+            if (model.getModelState() != MLModelState.DEPLOYED &&
+                    model.getModelState() != MLModelState.PARTIALLY_DEPLOYED) {
+                // if we find a model in the index but the model is not loaded into memory then we will
+                // load the model in memory
+                deployModel(modelId, ActionListener.wrap(deployModelResponse -> modelId = getTask(deployModelResponse.getTaskId()).getModelId(), e -> log.error("Metrics correlation model didn't get deployed to the index successfully", e)));
+            }
+        }
+
+        //We will be waiting here until actionListeners set the model id to the loadedModelId.
+        waitUntil(() -> {
+            if (modelId != null) {
+                MLModelState modelState = getModel(modelId).getModelState();
+                return modelState == MLModelState.DEPLOYED || modelState == MLModelState.PARTIALLY_DEPLOYED;
+            }
+            return false;
+        }, 120, TimeUnit.SECONDS);
+
+        Output djlOutput;
+        try {
+            if (modelId == null) {
+                throw new ExecuteException("Model is not loaded yet. Please try again.");
+            }
+            djlOutput = getPredictor().predict(processedInputData);
+        } catch (TranslateException translateException) {
+            throw new ExecuteException(translateException);
+        }
+
+        tensorOutputs.add(parseModelTensorOutput(djlOutput, null));
+        return new MetricsCorrelationOutput(tensorOutputs);
+    }
+
+    @VisibleForTesting
+    void searchModel(SearchRequest modelSearchRequest, ActionListener<Map<String, Object>> listener) {
+        client.execute(MLModelSearchAction.INSTANCE, modelSearchRequest, ActionListener.wrap(searchModelResponse -> {
+            Map<String, Object> modelInfo = null;
+            if (searchModelResponse != null) {
```

Comment:
What if searchModelResponse == null? We should return null `modelInfo` with `listener.onResponse(null)` ?

Reply:
Yeah we can send `null` and then in line 97, we can check `if (modelInfo == null || modelInfo.isEmpty()) {`